hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/HoodieDeleteBlock.java

```diff
@@ -16,64 +16,100 @@
 
 package com.uber.hoodie.common.table.log.block;
 
+import com.uber.hoodie.common.model.HoodieLogFile;
+import com.uber.hoodie.common.storage.SizeAwareDataInputStream;
+import com.uber.hoodie.exception.HoodieIOException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.FSDataInputStream;
+
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.nio.charset.Charset;
+import java.util.HashMap;
 import java.util.Map;
-
-import com.uber.hoodie.common.storage.SizeAwareDataInputStream;
-import org.apache.commons.lang3.StringUtils;
+import java.util.Optional;
 
 /**
  * Delete block contains a list of keys to be deleted from scanning the blocks so far
  */
 public class HoodieDeleteBlock extends HoodieLogBlock {
 
-  private final String[] keysToDelete;
+  private String[] keysToDelete;
 
-  public HoodieDeleteBlock(String[] keysToDelete, Map<LogMetadataType, String> metadata) {
-    super(metadata);
+  public HoodieDeleteBlock(String[] keysToDelete,
+                           Map<HeaderMetadataType, String> header) {
+    this(Optional.empty(), null, false, Optional.empty(), header, new HashMap<>());
     this.keysToDelete = keysToDelete;
   }
 
-  public HoodieDeleteBlock(String[] keysToDelete) {
-    this(keysToDelete, null);
+
+  private HoodieDeleteBlock(Optional<byte[]> content, FSDataInputStream inputStream,
+                            boolean readBlockLazily, Optional<HoodieLogBlockContentLocation> blockContentLocation,
+                            Map<HeaderMetadataType, String> header, Map<HeaderMetadataType, String> footer) {
+    super(header, footer, blockContentLocation, content, inputStream, readBlockLazily);
   }
 
   @Override
-  public byte[] getBytes() throws IOException {
+  public byte[] getContentBytes() throws IOException {
+
+    // In case this method is called before realizing keys from content
+    if (getContent().isPresent()) {
+      return getContent().get();
+    } else if (readBlockLazily && !getContent().isPresent() && keysToDelete == null) {
```

Comment:
does this code need to know about being lazy, given you pass in an Optional for content anyway? can't it just inflate if needed based on having content or not and return the value..

Reply:
Gave an explanation for this below.